cs130_back/handlers/group.go

```diff
@@ -62,18 +63,104 @@ func CreateGroup(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
 	}
 	defer r.Body.Close()
 
-	group := models.Group{}
+	group := models.Group{Name: p.Name, CourseID: p.CourseID, AdminID: p.AdminID}
 	if err := group.CreateGroup(db); err != nil {
 		respondWithError(w, http.StatusInternalServerError, err.Error())
 		return
 	}
 
-	if GetGroupByID(db, &group, w) == 0 {
+	if err := group.GetGroup(db); err != nil {
+		switch err {
+		case gorm.ErrRecordNotFound:
+			respondWithError(w, http.StatusNotFound, "Group not found")
+		default:
+			respondWithError(w, http.StatusInternalServerError, err.Error())
+		}
 		return
 	}
 
 	var gr CreateGroupResponse
 	populateGroupResponse(&group, &gr)
 
 	respondWithJSON(w, http.StatusCreated, gr)
-}
\ No newline at end of file
+}
+
+// GetGroup retrieves and returns the group
+func GetGroup(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
+	vars := r.URL.Query()
+	id, ok := strconv.Atoi(vars["g_id"][0])
+	if ok != nil {
+		respondWithError(w, http.StatusBadRequest, "Invalid group id")
+		return
+	}
+	group := models.Group{ID: id}
+	if GetGroupByID(db, &group, w) == 0 {
```

Comment:
For a group, you need to get their actual members` User objects as well. You also need to get the meeting time for this group.